alacritty/src/cli.rs

```diff
@@ -92,7 +92,6 @@ impl Options {
             config.ipc_socket |= self.socket.is_some();
         }
 
-        config.window.dynamic_title &= self.window_options.window_identity.title.is_none();
```

Comment:
I'm not 1000% sure removing this is safe, but it seems like the `preserve_title` option on `WindowContext` should do the same thing this operation did in the past. So I believe this should be redundant.

The advantage of not doing it here is that if we do it on the config directly, we have no way to distinguish between having it set in the config and having it set on the initial CLI. While storing the two separately allows us to make that distinction.